communication/src/coap_channel.cpp

```diff
@@ -124,10 +131,13 @@ ProtocolError CoAPMessageStore::receive(Message& msg, Channel& channel, system_t
 	if (msgtype==CoAPType::ACK || msgtype==CoAPType::RESET)
 	{
 		message_id_t id = msg.get_id();
+		CoAPMessage* coap_msg = from_id(id);
+		if (coap_msg) {
+			g_coapRoundTripMSec = time - coap_msg->get_send_time();
```

Comment:
Will this be safe for rollovers?

Reply:
Correct. This is associated with `SimpleUnsignedIntegerDiagnosticData` which now handles rollovers... 
This is the PR where rollovers are added - https://github.com/particle-iot/device-os/pull/2043